app/views/declaration/additionaldeclarationtype/declaration_type.scala.html

```diff
@@ -27,9 +27,10 @@
     title = messages("declaration.declarationType.title")
 ) {
 
-    @components.back_link(controllers.declaration.routes.DispatchLocationController.displayPage(mode), mode)
+    @components.back_link(controllers.declaration.routes.DispatchLocationController.displayPage(), request.mode)
 
-    @helper.form(controllers.declaration.routes.AdditionalDeclarationTypeController.submitForm(mode), 'autoComplete -> "off") {
+    @defining(controllers.declaration.routes.AdditionalDeclarationTypeController.submitForm()){ call =>
```

Comment:
This is the thing that puts me off, this would be a /POST with a query parameter, which is a bit weird.

Reply:
I think the existing solution already does that

I think we just have to bare in mind that this is a UI and not a REST API, so the same rules of convention don’t necessarily apply